datafusion/src/physical_plan/sorts/external_sort.rs

```diff
@@ -353,12 +386,79 @@ pub struct ExternalSortExec {
     input: Arc<dyn ExecutionPlan>,
     /// Sort expressions
     expr: Vec<PhysicalSortExpr>,
-    /// Execution metrics
-    metrics: ExecutionPlanMetricsSet,
+    /// Containing all metrics set created for sort, such as all sets for `sort_merge_join`s
+    all_metrics: AggregatedMetricsSet,
     /// Preserve partitions of input plan
     preserve_partitioning: bool,
 }
 
+#[derive(Debug, Clone)]
+struct AggregatedMetricsSet {
+    intermediate: Arc<std::sync::Mutex<Vec<ExecutionPlanMetricsSet>>>,
+    final_: Arc<std::sync::Mutex<Vec<ExecutionPlanMetricsSet>>>,
+}
```

Comment:
Introduce `AggregatedMetricsSet` for `ExternalSortExec` since it may include multi partial sort. Each partial sort itself is `SortPreservingMergeStream` and has its own metrics set.